### PR TITLE
add bert examples

### PR DESCRIPTION
**Pull request by Wesley-Jzy**

### CODE REVIEW

File: language/bert/basic/model/layers/linear.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import torch
+import torch.nn as nn
+from torch.nn import Parameter
+import torch.nn.functional as F
+import torch.nn.init as init
+
+
+class Linear(nn.Module):
+    """Linear layer with column parallelism.
```
Comment (feifeibear) [inline language/bert/basic/model/layers/linear.py:9]: Does this Linear come with column parallel?

File: language/bert/basic/model/layers/pooler.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import torch
+import torch.nn as nn
+from .linear import Linear
```
Comment (feifeibear) [inline language/bert/basic/model/layers/pooler.py:3]: You rename the colossal linear as Colo_Linear. Distinguish it from the torch one.

File: language/bert/basic/model/layers/embedding.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import torch
```
Comment (feifeibear) [inline language/bert/basic/model/layers/embedding.py:1]: Does this file come from Pytorch? Why not just import a pytorch package.

File: language/bert/basic/model/layers/bert_layer.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    def forward(self, Q, K, V, attn_mask):
+        scores = torch.matmul(Q, K.transpose(-1, -2)) / math.sqrt(self.d_k)
+        scores.masked_fill_(attn_mask, -1e4)
+        attn = nn.Softmax(dim=-1)(scores)
+        context = torch.matmul(attn, V)
+        return context
+
+
+class MultiHeadAttention(nn.Module):
```
Comment (feifeibear) [inline language/bert/basic/model/layers/bert_layer.py:35]: what's the diff from your MultiHeadAttention with pytorch?

File: language/bert/basic/data_processor/datasets/bert_dataset.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        # Build the samples mapping.
+        self.samples_mapping = get_samples_mapping_(self.indexed_dataset,
+                                                    data_prefix,
+                                                    num_epochs,
+                                                    max_num_samples,
+                                                    self.max_seq_length - 3,  # account for added tokens,
+                                                    short_seq_prob,
+                                                    self.seed,
+                                                    self.name,
+                                                    self.binary_head)
```
Comment (FrankLeeeee) [inline language/bert/basic/data_processor/datasets/bert_dataset.py:63]: The dataset can reused the one in sequence parallel.

**Comment by FrankLeeeee** [comment]
Hi @Wesley-Jzy , can you provide a detailed readme instead of an empty one?

**Comment by Wesley-Jzy** [comment]
@FrankLeeeee Thanks for review. It's still on the way since I try to fathom how to change model for TP and PP features now. I will add detailed README and delete code duplication in the final submit.

**Pull request closed without merging**